My Jetpack: Remove `Layout` component

renatoagds opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Related with #22649 

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Remove `Layout` component.
* Apply it structure direct on `ProductInterstitial`.
* Apply it structure direct on `ConnectionScreen`.
* Use `AdminPage` on both.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
N/A

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Open `My Jetpack` page.
* Navigate to interstitial pages and connection page.

oskosk commented:
Can you clarify why this is needed? Mentions a relation with https://github.com/Automattic/jetpack/issues/22649 and I see it moving things around but the relation is not evident to me

renatoagds commented:
@oskosk `Layout` component was created basically to reuse the footer and nav between interstitial and connection screens, as soon we moved Connection screen out of GoBackLink, this component was being used only to apply footer on both and navigation for Interstitials, so I just moved the logic of navigation back to Interstitial
and used `AdminPage` since it already implements the footer and is used for `MyJetpackScreen`.

renatoagds commented:
> Mentions a relation with 22649

@oskosk Already removed this relationship on description, it was the first idea but they're not related, updated it.

retrofox reviewed (changes requested):
It breaks the upgrade flow from the Interstitial Page, Renato.
You need to add the `select` dependency 

<img width="732" alt="image" src="https://user-images.githubusercontent.com/77539/156558690-a984fb75-41a5-42a9-94f3-be6e66c2b2cf.png">

Also, I think you should update with master branch.

renatoagds commented:
@retrofox It's already defined on line 6 of Interstitial 🤔.

I'll take a look again and try to reproduce.

retrofox commented:
> @retrofox It's already defined on line 6 of Interstitial 🤔.
> 
> I'll take a look again and try to reproduce.

Me too. Sorry about this :-/
